### PR TITLE
Add Shuryoken support to Appsignal

### PR DESCRIPTION
**Pull request by stijnster**
Hi guys,

here's a pull request for integrating appsignal and Shuryoken (https://github.com/phstc/shoryuken), using a hook and a middleware class.

I noticed that you ask to merge new features into the develop branch (readme file), but I'm under the impression that the develop branch is a bit behind. Therefor I chose to create this pull request against the master branch. Hope you don't mind.

### CODE REVIEW

**Comment by stijnster** [comment]
Oops, I see a lot of tests failing, due to older ruby integrations. Need to make my code backwards compatible. I'll come back on this soon ;)

**Comment by stijnster** [comment]
The build is still failing, but it's no longer due to my code;

> gems/ruby-1.9.3-p551/gems/capistrano-3.5.0/lib/capistrano/configuration.rb:132
> Ruby: 1.9.3 Gemfile: gemfiles/capistrano3.gemfile
> 
> syntax error, unexpected tLABEL (SyntaxError)
>     def install_plugin(plugin, load_hooks:true)

Looks like the capistrano gem is failing?

**Comment by matsimitsu** [comment]
Looks good, @stijnster! I've fixed the capistrano error on master ( https://github.com/appsignal/appsignal-ruby/commit/85f6e2e570ec89b519381910d6e2bad84c8a426a ), if you rebase, the tests should pass :)

**Comment by stijnster** [comment]
w00p w00p!

File: gemfiles/capistrano3.gemfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'capistrano', '>= 3.0'
+gem 'capistrano', '3.4.0'
```
Comment (thijsc) [inline gemfiles/capistrano3.gemfile]: Maybe go for '>~ 3.4' here.

File: spec/lib/appsignal/hooks/shoryuken_spec.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      Object.send(:remove_const, :Shoryuken)
+    end
+
+    its(:dependencies_present?) { should be_true }
+  end
+
+  context "without shoryuken" do
+    its(:dependencies_present?) { should be_false }
+  end
+end
```
Comment (thijsc) [inline spec/lib/appsignal/hooks/shoryuken_spec.rb]: We add a newline here.

File: lib/appsignal/hooks/shoryuken.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      def install
+        ::Shoryuken.configure_server do |config|
+          config.server_middleware do |chain|
+            chain.add Appsignal::Hooks::ShoryukenMiddleware
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (thijsc) [inline lib/appsignal/hooks/shoryuken.rb]: Newline here

File: spec/lib/appsignal/hooks/shoryuken_spec.rb [hunk 4]
```diff
@@ -0,0 +1,97 @@
+require 'spec_helper'
+
```
Comment (thijsc) [inline spec/lib/appsignal/hooks/shoryuken_spec.rb]: Some extra whitespace

**Comment by thijsc** [comment]
Thanks, great pull! Added a few very minor comments on whitespace matters.

Could you rebase this to one commit with a clear message? I think then we can merge.

**Comment by stijnster** [comment]
Hi guys, ok great!

Not sure on the comments regarding whitespace; did you want me to do that, or is that something you will do upon final integration?

I've squashed the commits together and tests are currently running. But it seems fine!

**Comment by stijnster** [comment]
All green.

She's all yours, take her home Captains!

**Comment by thijsc** [comment]
I tried to squash everything back to one commit, but getting a lot of merge conflicts. Could you maybe turn this into a single commit?

**Comment by stijnster** [comment]
All my changes are squashed into commit; 776446f no?

**Comment by thijsc** [comment]
Did you force push? I still see 20 commits in this pull.

**Comment by thijsc** [comment]
Oh that commit is fine, I'll cherry pick it.

**Comment by thijsc** [comment]
It's now in the gem: https://github.com/appsignal/appsignal-ruby/commit/b8f578fbebb74ace416e97a0ea86cc96aae6081d

And has been released in `1.2.0.alpha.3`. @stijnster: Could you test it?

**Comment by stijnster** [comment]
My log was;

[2016-05-17T23:28:28 (process) #10363][INFO] Loading Rails (4.2.6) integration
[2016-05-17T23:28:29 (process) #10363][INFO] Config key 'name' is being overwritten
[2016-05-17T23:28:29 (process) #10363][INFO] Starting AppSignal 1.1.5 (/opt/rubies/ruby-2.2.4/lib/ruby/gems/2.2.0/gems/passenger-4.0.59/helper-scripts/rack-preloader.rb, Ruby 2.2.4, x86_64-linux)
[2016-05-17T23:28:29 (process) #10363][INFO] Installing net_http hook
[2016-05-17T23:28:29 (process) #10363][INFO] Installing passenger hook

But is now;

[2016-06-01T12:39:38 (process) #29413][INFO] Installing net_http hook
[2016-06-01T12:39:38 (process) #29413][INFO] Installing passenger hook
[2016-06-01T12:39:38 (process) #29413][INFO] Installing shoryuken hook

And the data is coming in, so looks good!

**Comment by thijsc** [comment]
Great! Closing this one, this will be released in 1.2.

**Pull request closed without merging**